### PR TITLE
Implemented builder methods for shader input/output

### PR DESCRIPTION
**Pull request by trenouf**
These new methods are not yet used. That will come in the next commit.

Change-Id: I5fc193e78bb38cedf0d2fe9233332be055ffc52b

### CODE REVIEW

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by trenouf** [comment]
PR #150 has some test failures, so I may need to modify this change here before it is landed.

**Comment by amdrexu** [comment]
OK. If you need a second review, Please update the commit and ping us.

**Comment by trenouf** [comment]
V9: Fixed use of BuiltInSubgroup* in TCS/TES/GS.

File: builder/llpcBuilderBuiltIns.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+BUILTIN(NumWorkgroups,                      24, N,     C,        v3i32) // Number of local workgroups in the dispatch
+BUILTIN(PatchVertices,                      14, N,     HD,       i32)   // Number of vertices in the input patch
+BUILTIN(PointCoord,                         16, N,     P,        v2f32) // Coord of current fragment within the point
+BUILTIN(PointSize,                           1, MVHDG, HDG,      f32)   // Size of point primitives
+BUILTIN(Position,                            0, MVHDG, HDG,      v4f32) // Vertex position
+BUILTIN(PrimitiveId,                         7, MG,    HDGP,     i32)   // Index of the current primitive
+BUILTIN(SampleId,                           18, N,     P,        i32)   // Sample ID
+BUILTIN(SampleMask,                         20, P,     P,        ai32)  // Sample mask coverage
+BUILTIN(SamplePosition,                     19, N,     P,        v2f32) // Sub-pixel position of the sample being shaded
+BUILTIN(SubgroupEqMask,                   4416, N,     TMVHDGPC, v4i32)
```
Comment (amdrexu) [inline builder/llpcBuilderBuiltIns.h]: Please add missing comments for those SubgroupXxMask.

**Comment by trenouf** [comment]
V10: Added comments for BuiltInSibgroup*Mask

**Comment by trenouf** [comment]
I will land this if PR #150 testing passes.

**Comment by amdrexu** [comment]
It's OK for me.

**Pull request merged**